gravitee-apim-gateway/gravitee-apim-gateway-tests-sdk/README.adoc

```diff
@@ -0,0 +1,247 @@
+                assertThat(response.headers().get("X-Mock-Policy")).isEqualTo("Passed through mock policy");
+                assertThat(response.headers().contains("X-Mock-Policy-Second")).isTrue();
+                assertThat(response.headers().get("X-Mock-Policy-Second")).isEqualTo("reqHeaderValue");
+                return true;
+            })
+            .assertNoErrors();
+
+        wiremock.verify(0, getRequestedFor(urlPathEqualTo("/endpoint")));
+    }
+}
+```
+
+=== Key components
+
+* `@GatewayTest`: marks the class as testable with the SDK. It will run the necessary extensions (see https://junit.org/junit5/docs/current/user-guide/#extensions[JUnit 5's extension model]) to run the gateway and initialize components.
+* `@DeployApi`: allows to deploy an api thanks to its definition. You can pass an array of apis to deploy. ⚠️ When deploying multiple apis, they have to be configured with different id and a different entrypoint
+** Annotated at class level, it will deploy the apis once for the gateway instance, it means all the method annotated with `@Test` will use the same deployed apis.
+** Annotated at method level, it will deploy the apis only for the lifetime of the test method. After its execution, the apis will be undeployed
+* `AbstractPolicyTest<MockPolicy, MockPolicyConfiguration>`: extends from this abstract class allows to automatically configure the policy you are working on. We will see more details about this abstract class but it helps you configure a lot of stuff to test your policy.
+* `WebClient client`: You can see it as a parameter of your test method. This https://vertx.io/docs/apidocs/io/vertx/reactivex/ext/web/client/WebClient.html[Vertx Reactive Webclient] is injected by the SDK and automatically configured to reach the test gateway.
+* `wiremock`: is an object inherited from `AbstractPolicyTest` allowing you to mock your endpoints behavior. It is configured automatically to be used by your deployed apis.
+
+
+== Documentation
+
+=== @GatewayTest
+
+This annotation is used to run the gateway.
+
+Internally, it runs a custom JUnit5 Extension responsible for starting an APIM Gateway on an available port, and deploying apis configured thanks to `@DeployApi` annotation.
+
+This annotation uses the default configuration provided by the link:./src/main/resources/gravitee-default[gravitee-default] folder of the SDK, containing the `gravitee.yml`.
+
+In your policy repository, you can override it passing the name of the folder you want to use in your `src/test/resources`:
+```java
+@GatewayTest(configFolder = "my-folder")
+```
+
+=== @DeployApi
+
+This annotation is used to deploy apis thanks to their api definition.
+
+IMPORTANT: Policies declared in the api definition can only be deployed if they are part of your repository. That means, there is the policy under test, and it's possible to configure a test policy to do some testing processing. We will see how to configure it in <<_abstractgatewaytest>>.
```

Comment:
I'm not sure to fully understand these sentences.

Reply:
see [e5b99f6](https://github.com/gravitee-io/gravitee-api-management/pull/1939/commits/e5b99f670590cbfb3e8d811e22901ee754869561)